validator/decorator.py

```diff
@@ -28,6 +34,29 @@ def wrap(function):
     return wrap
 
 
+def register_cleanup(cleanup):
+    """Register a cleanup function to be called at the end of every validation
+    task. Takes either a callable, or a class with a `cleanup` class method."""
+
+    if not callable(cleanup):
```

Comment:
Just a side note: if the class has a `__call__` method, it'll be detected as callable. Shouldn't be an issue for us though, but maybe add a comment?